### PR TITLE
No jira add trace pkg and update logging, response and util

### PR DESCRIPTION
**Pull request by adieljulio**
- Add trace pkg
- Update logging event readme for correct extractfunction
- Update response camelCase error response and change WriteErrorString to WriteHeaderAndJson 
- Update extract default util

@anggorodewanto @thoriqsatriya @fadhillahentino @rjjatson @hvlcrs @yani-satwib

### CODE REVIEW

File: pkg/trace/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Trace
+
+This package have filter for generate traceID if no traceID in request header.
```
Comment (anggorodewanto) [inline pkg/trace/README.md]: "This package contains filter for generating `traceID` if there is no `traceID` in request header."

File: pkg/trace/trace.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+)
+
+const (
+	traceIDKey = "X-Ab-TraceID"
+)
+
+func Filter() restful.FilterFunction {
+	return func(req *restful.Request, resp *restful.Response, chain *restful.FilterChain) {
+		if req.HeaderParameter(traceIDKey) == "" {
+			traceID, _ := generateUUID()
```
Comment (anggorodewanto) [inline pkg/trace/trace.go]: there's an unhandled error here

File: pkg/trace/README.md (hunk 1 above)
Comment (adieljulio) [reply pkg/trace/README.md]: Updated

File: pkg/trace/trace.go (hunk 2 above)
Comment (adieljulio) [reply pkg/trace/trace.go]: Updated, done handle error

**Pull request merged**